lib/instrumentation/openai.js

```diff
@@ -289,6 +298,11 @@ module.exports = function initialize(agent, openai, moduleName, shim) {
     openai.Embeddings.prototype,
     'create',
     function wrapEmbeddingCreate(shim, embeddingCreate, name, args) {
+      if (shouldSkipInstrumentation(agent.config, shim) === true) {
```

Comment:
same comment as in langchain. this is happening too early and it won't properly record the span

```
--- a/lib/instrumentation/openai.js
+++ b/lib/instrumentation/openai.js
@@ -298,11 +298,6 @@ module.exports = function initialize(agent, openai, moduleName, shim) {
     openai.Embeddings.prototype,
     'create',
     function wrapEmbeddingCreate(shim, embeddingCreate, name, args) {
-      if (shouldSkipInstrumentation(agent.config, shim) === true) {
-        shim.logger.debug('skipping sending of ai data')
-        return
-      }
-
       const [request] = args
       return new RecorderSpec({
         name: OPENAI.EMBEDDING,
@@ -315,10 +310,16 @@ module.exports = function initialize(agent, openai, moduleName, shim) {
             // In that case, we define it to be an empty object.
             response = {}
           }
+          segment.end()
+
+          if (shouldSkipInstrumentation(agent.config, shim) === true) {
+            shim.logger.debug('skipping sending of ai data')
+            return
+          }
+
           response.headers = segment[openAiHeaders]
           // explicitly end segment to get consistent duration
           // for both LLM events and the segment
-          segment.end()
           const embedding = new LlmEmbedding({
             agent,
             segment,
```

take note i moved ending segment before assigning the segment to a symbol on headers